src/test/components/NetworkChart.test.js

```diff
@@ -130,6 +156,38 @@ describe('NetworkChart', function() {
     expect(container.firstChild).toMatchSnapshot();
     expect(drawCalls).toMatchSnapshot();
   });
+
+  it('displays a context menu when right clicking', () => {
+    // Context menus trigger asynchronous operations for some behaviors, so we
+    // use fake timers to avoid bad interactions between tests.
+    jest.useFakeTimers();
+
+    const markers = [
+      ...getNetworkMarkers({
+        uri: 'https://mozilla.org/1',
+        id: 1,
+        startTime: 10,
+        endTime: 60,
+      }),
+      ...getNetworkMarkers({
+        uri: 'https://mozilla.org/2',
+        id: 2,
+        startTime: 20,
+        endTime: 70,
+      }),
+    ];
+    const { getByText, getContextMenu, rightClick } = setupWithPayload(markers);
+    rightClick(getByText('/1'));
+
+    expect(getContextMenu()).toHaveClass('react-contextmenu--visible');
```

Comment:
Optional nudge: It'd be nice to not have to rely on class names, since they are a bit brittle when changing implementation details, although that may be technically hard.

Reply:
But the class is what makes it visible in the end, so I think it's the best thing to test in the test as the user-visible change... even if of course we'll have to change it if the class ever changes.